katello/katello-change-hostname

```diff
@@ -50,108 +124,132 @@ def yesno
+  trusted_hosts = fpc_answers["foreman_proxy"]["trusted_hosts"]
+  oauth_consumer_key = fpc_answers["foreman_proxy"]["oauth_consumer_key"]
+  oauth_consumer_secret = fpc_answers["foreman_proxy"]["oauth_consumer_secret"]
+  pulp_oauth_secret = fpc_answers["foreman_proxy_content"]["pulp_oauth_secret"]
+  certs_tar = @options[:certs_tar]
+  foreman_url = fpc_answers["puppet"]["server_foreman_url"]
+
+  fpc_installer_args = " --foreman-proxy-content-parent-fqdn #{parent_fqdn}" \
+  " --foreman-proxy-register-in-foreman #{register_in_foreman}" \
+  " --foreman-proxy-foreman-base-url #{base_url}" \
+  " --foreman-proxy-oauth-consumer-key #{oauth_consumer_key}" \
+  " --foreman-proxy-oauth-consumer-secret #{oauth_consumer_secret}" \
+  " --foreman-proxy-content-pulp-oauth-secret #{pulp_oauth_secret}" \
+  " --foreman-proxy-content-certs-tar #{certs_tar}" \
+  " --puppet-server-foreman-url #{foreman_url}" \
+  " --foreman-proxy-register-in-foreman false"
+
+  trusted_hosts.each { |host| fpc_installer_args << " --foreman-proxy-trusted-hosts #{host}" }
+
+  fpc_installer_args
+end
 
-  opt.on("-u","--username USERNAME","admin username (required)") do |username|
-    options[:username] = username
-  end
+def succesful_hostname_change_message
+  STDOUT.puts "**** Hostname change complete! ****"
+  if @foreman_proxy_content
+    STDOUT.puts "IMPORTANT: You will have to update the Name and URL of the Smart Proxy in #{@options[:program].capitalize} to the new hostname."
+  else
+    STDOUT.puts "
 
-  opt.on("-p","--password PASSWORD","admin password (required)") do |password|
-    options[:password] = password
-  end
+    You will need to re-register any #{@plural_proxy} or clients with the server.
 
-  opt.on("-g","--program PROGRAM","name of the program you are modifying (defaults to #{DEFAULT_PROGRAM})") do |program|
-    options[:program] = program
-  end
+    When re-registering clients, you will have to reinstall the bootstrap RPM.
+    Any mention of the #{old_hostname} on your clients will have to change to #{new_hostname} (i.e. /etc/hosts)
 
-  opt.on("-S","--scenario SCENARIO","name of the scenario you are modifying (defaults to #{DEFAULT_SCENARIO})") do |scenario|
-    options[:scenario] = scenario
-  end
+    For #{@plural_proxy}, you will need to regenerate the certs on the #{@options[:scenario].capitalize} server and reinstall the #{@proxy}
 
+    If you want to use custom certificates, re-run the #{@options[:program]}-installer with custom certificate @options
 
-  if has_disable_system_check_option?
-    opt.on("-d","--disable-system-checks","runs the installer with --disable-system-checks") do |system_check|
-      options[:system_check] = true
-    end
+    Short hostnames have not been updated, please update those manually.
+    "
   end
+  exit(true)
+end
 
-  opt.on("-h","--help","help") do
-    puts opt_parser
-    exit
+def warning_message
+  STDOUT.print("This script will modify your system.")
+  unless @foreman_proxy_content
```

Comment:
Clients can register through a smart proxy as well and would need to do the update below.